### PR TITLE
Display input value description

### PR DESCRIPTION
**Pull request by tristetemps666**

### CODE REVIEW

File: Nodes/Voronoi.clbnode [hunk 1]
```diff
@@ -1,4 +1,4 @@
-INPUT int `Grid Size`;
+INPUT int `Grid Size`; /// ceci est un commentaire
```
Comment (JulesFouchy) [inline Nodes/Voronoi.clbnode:1]: To remove

File: src/Module_Nodes/NodeInputDefinition.h [hunk 2]
```diff
@@ -1,11 +1,14 @@
 #pragma once
+#include <string>
 #include "FunctionSignature.h"
 
 namespace Lab {
 
 struct NodeInputDefinition_Data {
     std::string       name{};
     FunctionSignature signature{};
+    std::string       description{"cacaboudin"};
```
Comment (JulesFouchy) [inline src/Module_Nodes/NodeInputDefinition.h:10]: To remove

File: src/Module_Nodes/NodeInputDefinition.h [hunk 3]
```diff
@@ -14,6 +17,7 @@ class NodeInputDefinition {
 
     auto name() const -> auto const& { return _data.name; }
     auto signature() const -> auto const& { return _data.signature; }
+    void set_description(const std::string& description) { _data.description = description; }
```
Comment (JulesFouchy) [inline src/Module_Nodes/NodeInputDefinition.h:20]: Put const on the right. Here, and in many other places in this pull request.

File: src/Module_Nodes/parse_node_definition.cpp [hunk 4]
```diff
@@ -1,7 +1,9 @@
 #include "parse_node_definition.h"
 #include <Cool/Expected/RETURN_IF_UNEXPECTED.h>
 #include <Cool/String/String.h>
+#include <vcruntime.h>
```
Comment (JulesFouchy) [inline src/Module_Nodes/parse_node_definition.cpp:4]: To remove

File: src/Module_Nodes/parse_node_definition.cpp [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
 #include <algorithm>
+#include <cstdio>
```
Comment (JulesFouchy) [inline src/Module_Nodes/parse_node_definition.cpp:6]: To remove

File: src/Module_Nodes/parse_node_definition.cpp [hunk 6]
```diff
@@ -431,6 +433,99 @@ static auto find_outputs(std::string const& text, NodeDefinition_Data& res)
     );
 }
 
+template<typename T>
+void modify_description(const std::string& name_property, Cool::InputDefinition<T>& input, const std::string& param_description)
```
Comment (JulesFouchy) [inline src/Module_Nodes/parse_node_definition.cpp:437]: More explicit name for the function

File: src/Module_Nodes/parse_node_definition.cpp [hunk 7]
```diff
@@ -431,6 +433,99 @@ static auto find_outputs(std::string const& text, NodeDefinition_Data& res)
...
+{
+    if (input.name == name_property)
+    {
+        input.description = param_description;
+    }
+}
+
+std::string get_rid_of_triple_slashs(const std::string& text, const size_t& last)
```
Comment (JulesFouchy) [inline src/Module_Nodes/parse_node_definition.cpp:445]: "slashes" instead of "slashs"

File: src/Module_Nodes/parse_node_definition.cpp [hunk 8]
```diff
@@ -459,6 +556,8 @@ auto parse_node_definition(std::filesystem::path filepath, std::string text)
             return tl::make_unexpected(*err);
     }
 
+    auto a = find_properties_descriptions(text_with_comments, res);
```
Comment (JulesFouchy) [inline src/Module_Nodes/parse_node_definition.cpp:559]: Handle error if any

File: src/Module_Nodes/parse_node_definition.cpp [hunk 9]
```diff
@@ -441,6 +536,8 @@ auto parse_node_definition(std::filesystem::path filepath, std::string text)
 
     NodeDefinition_Data res{};
 
+    std::string text_with_comments = text;
```
Comment (JulesFouchy) [inline src/Module_Nodes/parse_node_definition.cpp:539]: const

File: src/Module_Nodes/parse_node_definition.cpp (hunk 7 above)
Comment (tristetemps666) [reply src/Module_Nodes/parse_node_definition.cpp:445]: I just deleted the function wich is is no longer usefull, we do it directly

**Pull request merged**